### PR TITLE
Fetch private submissions after polling when checkbox ticked

### PR DESCRIPTION
**Pull request by gautamjajoo**
When `Include Private Submission` is checked the private submissions are not fetched after 5seconds of polling. 
Also, when the leaderboard has all private submissions, previously no results are shown, but now the checkbox is also visible.

@Ram81 @Kajol-Kumari

### CODE REVIEW

File: frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html [hunk 1]
```diff
@@ -87,14 +90,15 @@ <h5 class="fw-light">Leaderboard</h5>
...
-              <mat-checkbox
-                class="pull-right complete-leaderboard"
-                *ngIf="isChallengeHost && showLeaderboardToggle"
-                (change)="toggleLeaderboard(getAllEntries)"
-                [(ngModel)]="getAllEntries"
-              >
-                <label class="fw-light complete-leaderboard-text"> {{ getAllEntriesTextOption }}</label>
-              </mat-checkbox>
+              </div>
+                <mat-checkbox
```
Comment (Kajol-Kumari) [inline frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html:94]: @gautamjajoo  This checkbox will appear if we have total submission `0` We need a check here that only if my total submission(both public and private) is >0, then we should show this checkbox

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]

File: frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html (hunk 1 above)
Comment (gautamjajoo) [reply frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html:94]: Fixed. Now the checkbox only appears when `public + private > 0`.

**Review by Kajol-Kumari — APPROVED** [review]
LGTM 👍

**Review by Ram81 — APPROVED** [review]
LGTM

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**